src/Compilers/CSharp/Portable/Symbols/Extensions/SynthesizedExtensionMarker.cs

```diff
@@ -74,6 +75,11 @@ protected override (TypeWithAnnotations ReturnType, ImmutableArray<ParameterSymb
 
                 ParameterSymbol? parameter = ParameterHelpers.MakeExtensionReceiverParameter(withTypeParametersBinder: signatureBinder, owner: this, parameterList, diagnostics);
 
+                if (parameter is { })
+                {
+                    checkUnderspecifiedGenericExtension(parameter.Type, ContainingType.TypeParameters, diagnostics, parameter.GetFirstLocation());
```

Comment:
>parameter.GetFirstLocation() [](http://example.com/codeflow?start=116&length=28)

It is preferred to not create locations until after we know that it will be used. It looks like in this case we can easily achieve that by passing ```parameter``` instead of its type and location.